### PR TITLE
[SPARK-42644][INFRA] Add `hive` dependency to `connect` module

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to update `connect` module dependency from `sql` to hive.

### Why are the changes needed?

SPARK-41725 added `hive` dependency via test suite.

https://github.com/apache/spark/blob/41d3103f4d69a9ec25d9f78f3f94ff5f3b64ef78/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala#L75

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @grundprinzip , @hvanhovell , @zhengruifeng, @HyukjinKwon , @xinrong-meng

**Review by HyukjinKwon — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you! Let me fix the doctest too.

File: dev/sparktestsupport/utils.py [hunk 1]
```diff
@@ -109,11 +109,11 @@ def determine_modules_to_test(changed_modules, deduplicated=True):
...
     ['graphx', 'examples']
-    >>> [x.name for x in determine_modules_to_test([modules.sql])]
+    >>> sorted([x.name for x in determine_modules_to_test([modules.sql])])
     ... # doctest: +NORMALIZE_WHITESPACE
-    ['sql', 'avro', 'connect', 'docker-integration-tests', 'hive', 'mllib', 'protobuf',
-     'sql-kafka-0-10', 'examples', 'hive-thriftserver', 'pyspark-sql', 'repl', 'sparkr',
-     'pyspark-connect', 'pyspark-mllib', 'pyspark-pandas', 'pyspark-pandas-slow', 'pyspark-ml']
+    ['avro', 'connect', 'docker-integration-tests', 'examples', 'hive', 'hive-thriftserver',
+     'mllib', 'protobuf', 'pyspark-connect', 'pyspark-ml', 'pyspark-mllib', 'pyspark-pandas',
+     'pyspark-pandas-slow', 'pyspark-sql', 'repl', 'sparkr', 'sql', 'sql-kafka-0-10']
```
Comment (dongjoon-hyun) [inline dev/sparktestsupport/utils.py:116]: The content is the same. Only sorted.

**Comment by LuciferYang** [comment]
Hi ~ @dongjoon-hyun , could you please help check the results of local execution of 

```
build/sbt clean "connect-client-jvm/test" -Dspark.debug.sc.jvm.client=true
```
?
I found the following errors:
```
[info] ClientE2ETestSuite:
Starting the Spark Connect Server...
Using jar: /Users/yangjie01/SourceCode/git/spark-mine-sbt/connector/connect/server/target/scala-2.12/spark-connect-assembly-3.5.0-SNAPSHOT.jar
Ready for client connections.
java.lang.RuntimeException: Failed to start the test server on port 15971.
  | => cat org.apache.spark.sql.connect.client.util.RemoteSparkSession.beforeAll(RemoteSparkSession.scala:129)
	at org.apache.spark.sql.connect.client.util.RemoteSparkSession.beforeAll$(RemoteSparkSession.scala:120)
	at org.apache.spark.sql.ClientE2ETestSuite.beforeAll(ClientE2ETestSuite.scala:36)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:212)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at org.apache.spark.sql.ClientE2ETestSuite.run(ClientE2ETestSuite.scala:36)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
	at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
....

Suppressed: io.grpc.StatusRuntimeException: INTERNAL: Error while instantiating 'org.apache.spark.sql.hive.HiveSessionStateBuilder':
		at io.grpc.Status.asRuntimeException(Status.java:535)
		at io.grpc.stub.ClientCalls$BlockingResponseStream.hasNext(ClientCalls.java:660)
		at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:45)
		at scala.collection.Iterator.find(Iterator.scala:993)
		at scala.collection.Iterator.find$(Iterator.scala:992)
		at scala.collection.AbstractIterator.find(Iterator.scala:1431)
		at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:133)
		at org.apache.spark.sql.SparkSession.$anonfun$sql$1$adapted(SparkSession.scala:125)
		at org.apache.spark.sql.SparkSession.newDataset(SparkSession.scala:258)
		at org.apache.spark.sql.SparkSession.newDataFrame(SparkSession.scala:252)
		at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:125)
		at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:109)
		at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:147)
		at org.apache.spark.sql.connect.client.util.RemoteSparkSession.beforeAll(RemoteSparkSession.scala:135)
		... 13 more
```

Seems also related to [SPARK-41725](https://issues.apache.org/jira/browse/SPARK-41725)?

**Comment by dongjoon-hyun** [comment]
Yes, you need `-Phive` after SPARK-41725.  I already checked that. I made this PR because of that, @LuciferYang .

**Comment by LuciferYang** [comment]
> Yes, you need `-Phive` after [SPARK-41725](https://issues.apache.org/jira/browse/SPARK-41725). I already checked that. I made this PR because of that, @LuciferYang .

Thanks ~

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and @LuciferYang . Merged to master/3.4.

**Comment by zhengruifeng** [comment]
Late LGTM, thank you @dongjoon-hyun !

**Pull request closed without merging**